addon/adapters/interop-debug-adapter.js

```diff
@@ -0,0 +1,70 @@
+import { DebugAdapter } from 'ember-data/-private';
+import M3DebugAdapter from './m3-debug-adapter';
+import { get, defineProperty } from '@ember/object';
+import { inject } from '@ember/service';
+import { default as MegamorphicModel } from '../model';
+
+/*
+  Extend `Ember.DataAdapter` with m3 specific code
+
+  @class DebugAdapter
+  @extends Ember.DataAdapter
+  @private
+*/
+
+export default class InteropDebugAdapter extends DebugAdapter {
+  init() {
+    super.init(...arguments);
+    const store = get(this, 'store');
+    const schema = get(this, 'schema');
+    this._m3DebugAdapter = M3DebugAdapter.create({
+      store,
+      schema,
+    });
+  }
+
+  /**
+    Function to call m3's debug adapter addedType method
+    This is only needed for m3 records that have been dynamically added after
+    watchModelTypes has been called
+    @private
+    @method addedType
+    @param {String} type Model type
+  */
+  addedType(type) {
+    this._m3DebugAdapter.addedType(type);
+  }
+
+  /**
+    Calls the getRecordColumnValues function from either m3's debug adapter
+    or Ember Data's debug adapter depending on the record type
+    @public
+    @method getRecordColumnValues
+    @param {MegamorphicModel|DS.Model} record to get values from
+    @return {Object} Keys should match column names defined by the model type
+  */
+  getRecordColumnValues(record) {
+    if (record instanceof MegamorphicModel) {
+      return this._m3DebugAdapter.getRecordColumnValues(record);
+    }
+    return super.getRecordColumnValues(record);
+  }
+
+  /**
+    Fetch the model types and observe them for changes
+    Also keeps track of what model types have been handled by the adapter
+    @public
+    @method watchModelTypes
+    @param {Function} typesAdded Callback to call to add types
+    Takes an array of objects containing wrapped types (returned from `processAddedTypes`)
+    @param {Function} typesUpdated Callback to call when a type has changed
+    Takes an array of objects containing wrapped types
+    @return {Function} Method to call to remove all observers
+  */
+  watchModelTypes(typesAdded, typesUpdated) {
+    this._m3DebugAdapter.watchModelTypes(typesAdded, typesUpdated);
```

Comment:
I'm uncertain as to how I should handle combining the outputs of the release methods, since each of the `watchModelTypes` returns a callback but I think there should be a shared array of release functions to iterate over for both:
<img width="542" alt="Screen Shot 2019-09-09 at 9 47 36 AM" src="https://user-images.githubusercontent.com/10149117/64550656-3716d680-d2e8-11e9-9818-742546361038.png">

It's unclear to me if I can accomplish this without tweaking how `watchModelTypes` works in [Ember's Data Adapter](https://github.com/emberjs/ember.js/blob/master/packages/@ember/-internals/extension-support/lib/data_adapter.js#L133). I can, of course, change the logic for the `m3's` adapter as well but need some guidance as to what the path forward should be.

Reply:
Return a function that invokes both callbacks

Reply:
```js
watchModelTypes(typesAdded, typesUpdated) {
  let releaseM3 = this._m3DebugAdapter.watchModelTypes(typesAdded, typesUpdated);
  let releaseSuper = super.watchModelTypes(typesAdded, typesUpdated);

  return () => {
    releaseSuper();
    releaseM3();
  }
}
```

Reply:
Ohhh, ok seems obvious now that you said it 🤦‍♀Will do that, thanks!

Reply:
np! 😀

I wouldn't worry about catching errors or anything unless we really need to ensure `releaseM3` runs or something; i would expect a simple composition here should be fine and let any errors get thrown.